Gloo Mesh Enterprise v1.1.0 Upgrade docs (#1846)
solo-io/gloo-mesh: docs/content/operations/upgrading.md

---
title: Upgrading Gloo Mesh Enterprise
menuTitle: Upgrading Gloo Mesh Enterprise
weight: 100
description: Best practices for upgrading Gloo Mesh Enterprise
---

{{< notice warning >}}

> This upgrade process is for Gloo Mesh Enterprise, from v1.1.0-betax to v1.1.0.

Reviewer: Would this upgrade process cause any downtime, like the deployments would go down or other types of impact? If so, could we call that out?

Author: This _should_ not cause any downtime, given that we are rolling over the management plane, and that does not affect traffic. @ilackarms is this correct, or should we still add a warning?

Reviewer 2: The data plane stays up but you can have some down time of the management plane, which could result in delays of new configs being applied, which could potentially cause some downtime in certain situations.

I think we should keep the "not guaranteed zero downtime" disclaimer for now - but true zero downtime upgrades will be a high priority item for the next release. Happy to discuss further internally, if needed!

Reviewer: ok, how about:

```suggestion
This upgrade process is for Gloo Mesh Enterprise, from v1.1.0-betax to v1.1.0. During the upgrade, the data plane continues to run, but you might not be able to modify the configurations through the management plane. Because zero downtime is not guaranteed, try testing the upgrade in a staging environment before upgrading your production environment.
```

Reviewer 2: Perfect!

> If you have an older version installed, the recommended way to access new functionality
> is via a fresh install of Gloo Mesh Enterprise.

Reviewer: Does this mean uninstall and reinstall? If so, maybe rephrase:
```suggestion
For previous versions, uninstall and re-install Gloo Mesh Enterprise, such as described in one of the [setup guides]({{% versioned_link_path fromRoot="/getting_started//#deploying-gloo-mesh" %}}).
```

{{< /notice >}}

> 1\. 

Reviewer: Maybe describe why they do this, like:
```suggestion
1\. In your terminal, set environment variables for your current and target Gloo Mesh Enterprise versions to use in the subsequent steps.
```

```shell
# Set your current version here
CURRENT_VERSION=v1.1.0-beta35
# Add your desired version here
UPGRADE_VERSION=v1.1.0
```

> 2\. Upgrade CRDs on your management cluster and all of your data plane clusters:

Reviewer: Do they have to switch the kube contexts to each individual cluster to run these commands?
```suggestion
2\. Upgrade the Gloo Mesh Enterprise CRDs on your management cluster and all of your data plane clusters. You can change contexts before applying the CRDs by running `kubectl config set-context <cluster-name>`.
```

```shell

> k apply -f https://raw.githubusercontent.com/solo-io/gloo-mesh/$UPGRADE_VERSION/install/helm/gloo-mesh-crds/crds/admin.enterprise.mesh.gloo.solo.io_v1alpha1_crds.yaml
> k apply -f https://raw.githubusercontent.com/solo-io/gloo-mesh/$UPGRADE_VERSION/install/helm/gloo-mesh-crds/crds/discovery.mesh.gloo.solo.io_v1_crds.yaml
> k apply -f https://raw.githubusercontent.com/solo-io/gloo-mesh/$UPGRADE_VERSION/install/helm/gloo-mesh-crds/crds/multicluster.solo.io_v1alpha1_crds.yaml
> k apply -f https://raw.githubusercontent.com/solo-io/gloo-mesh/$UPGRADE_VERSION/install/helm/gloo-mesh-crds/crds/networking.enterprise.mesh.gloo.solo.io_v1beta1_crds.yaml
> k apply -f https://raw.githubusercontent.com/solo-io/gloo-mesh/$UPGRADE_VERSION/install/helm/gloo-mesh-crds/crds/networking.mesh.gloo.solo.io_v1_crds.yaml
> k apply -f https://raw.githubusercontent.com/solo-io/gloo-mesh/$UPGRADE_VERSION/install/helm/gloo-mesh-crds/crds/observability.enterprise.mesh.gloo.solo.io_v1_crds.yaml
> k apply -f https://raw.githubusercontent.com/solo-io/gloo-mesh/$UPGRADE_VERSION/install/helm/gloo-mesh-crds/crds/rbac.enterprise.mesh.gloo.solo.io_v1_crds.yaml
> k apply -f https://raw.githubusercontent.com/solo-io/gloo-mesh/$UPGRADE_VERSION/install/helm/gloo-mesh-crds/crds/settings.mesh.gloo.solo.io_v1_crds.yaml

Reviewer: I know this process might change, but maybe just spell out `kubectl` in case they don't have a shorthand set.
```suggestion
kubectl apply -f https://raw.githubusercontent.com/solo-io/gloo-mesh/$UPGRADE_VERSION/install/helm/gloo-mesh-crds/crds/admin.enterprise.mesh.gloo.solo.io_v1alpha1_crds.yaml
kubectl apply -f https://raw.githubusercontent.com/solo-io/gloo-mesh/$UPGRADE_VERSION/install/helm/gloo-mesh-crds/crds/discovery.mesh.gloo.solo.io_v1_crds.yaml
kubectl apply -f https://raw.githubusercontent.com/solo-io/gloo-mesh/$UPGRADE_VERSION/install/helm/gloo-mesh-crds/crds/multicluster.solo.io_v1alpha1_crds.yaml
kubectl apply -f https://raw.githubusercontent.com/solo-io/gloo-mesh/$UPGRADE_VERSION/install/helm/gloo-mesh-crds/crds/networking.enterprise.mesh.gloo.solo.io_v1beta1_crds.yaml
kubectl apply -f https://raw.githubusercontent.com/solo-io/gloo-mesh/$UPGRADE_VERSION/install/helm/gloo-mesh-crds/crds/networking.mesh.gloo.solo.io_v1_crds.yaml
kubectl apply -f https://raw.githubusercontent.com/solo-io/gloo-mesh/$UPGRADE_VERSION/install/helm/gloo-mesh-crds/crds/observability.enterprise.mesh.gloo.solo.io_v1_crds.yaml
kubectl apply -f https://raw.githubusercontent.com/solo-io/gloo-mesh/$UPGRADE_VERSION/install/helm/gloo-mesh-crds/crds/rbac.enterprise.mesh.gloo.solo.io_v1_crds.yaml
kubectl apply -f https://raw.githubusercontent.com/solo-io/gloo-mesh/$UPGRADE_VERSION/install/helm/gloo-mesh-crds/crds/settings.mesh.gloo.solo.io_v1_crds.yaml
```

Author: Oh yeah absolutely - we definitely should never use `k` in our docs.

```

3\. Clean up resources that Helm does not manage.
```shell
kubectl delete rolebinding -n gloo-mesh enterprise-networking-test enterprise-agent-test
kubectl delete pod -n gloo-mesh enterprise-agent-test
```

> 4\. Helm Upgrade for the Management Plane

Reviewer: Maybe 
```suggestion
4\. Set your Kubernetes context to the management plane cluster, and upgrade the Helm installation.
```

```shell
helm upgrade --install gloo-mesh --namespace gloo-mesh \
  'https://storage.googleapis.com/gloo-mesh-enterprise/gloo-mesh-enterprise/gloo-mesh-enterprise-1.1.0.tgz' \
  --set licenseKey=$LICENSE_KEY
```

> 5\. Helm Upgrade for the Data Planes

Reviewer: ```suggestion
5\. For each data plane cluster, set your Kubernetes context to the cluster, and upgrade the Helm installation.
```

```shell
helm upgrade --install enterprise-agent --namespace gloo-mesh https://storage.googleapis.com/gloo-mesh-enterprise/enterprise-agent/enterprise-agent-1.1.0.tgz
```

> 6\. Run `meshctl check` on the management to verify that all resources are in a healthy state. Refer to our

Reviewer: Are there any prereqs they have to do to get `meshctl check` to work? It doesn't work for me.
```suggestion
6\. Set the Kubernetes context to the management plane cluster and then check that your Gloo Mesh Enterprise resources are in a healthy state. Refer to our
```

Author: They need to have a `meshctl` version that is v1.1.0 or higher.

---
title: Upgrading Gloo Mesh Enterprise
menuTitle: Upgrading Gloo Mesh Enterprise
weight: 100
description: Best practices for upgrading Gloo Mesh Enterprise
---

{{< notice warning >}}
This upgrade process is for Gloo Mesh Enterprise, from v1.1.0-betax to v1.1.0. During the upgrade, the data plane continues to run, but you might not be able to modify the configurations through the management plane. Because zero downtime is not guaranteed, try testing the upgrade in a staging environment before upgrading your production environment.

> If you are on a previous version, uninstall and re-install Gloo Mesh Enterprise, as described in one of the [setup guides]({{% versioned_link_path fromRoot="/getting_started//#deploying-gloo-mesh" %}}).

Reviewer: I saw in the preview that I messed up the link :( 
```suggestion
If you are on a previous version, uninstall and re-install Gloo Mesh Enterprise, as described in one of the [setup guides]({{% versioned_link_path fromRoot="/getting_started/#deploying-gloo-mesh" %}}).
```

{{< /notice >}}

1\. In your terminal, set environment variables for your current and target Gloo Mesh Enterprise versions to use in the subsequent steps.

```shell
# Set your current version here (Must be a v1.1.0-betax version)
CURRENT_VERSION=v1.1.0-beta35
# Add your desired version here (Must be a v1.1.x version)
UPGRADE_VERSION=v1.1.0
```

2\. Upgrade the Gloo Mesh CRDs on your management cluster and all of your data plane clusters. You can change contexts before applying the CRDs by running `kubectl config set-context <cluster-name>`.
```shell

> kubectl apply -f https://raw.githubusercontent.com/solo-io/gloo-mesh/$UPGRADE_VERSION/install/helm/gloo-mesh-crds/crds/admin.enterprise.mesh.gloo.solo.io_v1alpha1_crds.yaml

Author: Caveat: we actually don't want to use the $UPGRADE_VERSION here (It's fine for the special v1.1.0 case because our releases line up correctly).

We actually want to use the `gloo-mesh` dependency version in the gloo-mesh-enterprise release that we are upgrading to. This is information that is only available in the `go.mod` file (in a private repository) or by searching through the changelogs (should be available in the docs, but is currently broken) for the latest Dependency Bump for gloo-mesh.
